### ISSUE TITLE
Upgrade to v6 async/await

### ISSUE THREAD

**Issue by mauroservienti** [Particular/NServiceBus.CastleWindsor#4]
Related to https://github.com/Particular/NServiceBus/issues/3076

**Comment by johnsimons** [comment]
@SimonCropp it looks like you have done it ?

**Comment by danielmarbach** [comment]
@johnsimons I want to ditch the ATTs and write a few tests for castle itself

**Comment by SimonCropp** [comment]
Then this should not be flagged with v6

**Comment by mauroservienti** [comment]
Why @SimonCropp ?

**Comment by SimonCropp** [comment]
@mauroservienti because re doing the tests when this already passes all the previous acceptance tests is no a blocker for v6

**Comment by mauroservienti** [comment]
So @danielmarbach open another issue and let's use this to keep track that we have not release the V6 compatible package, so this cannot be considered done.

**Comment by danielmarbach** [comment]
It doesn't use latest V6. While upgrading to latest V6 I wanted to ditch
the current ATTs and replace with a few smoke tests. I consider this V6

**Comment by johnsimons** [comment]
@danielmarbach are u working on this or can I  ?

**Comment by danielmarbach** [comment]
Yes I have it locally. Sorry I push today

**Comment by danielmarbach** [comment]
I'm currently blocked by https://github.com/Particular/NServiceBus/pull/3416

**Comment by danielmarbach** [comment]
https://github.com/Particular/NServiceBus/pull/3416 has been merged

**Comment by danielmarbach** [comment]
Done as part of https://github.com/Particular/NServiceBus.CastleWindsor/pull/6
